tests/compiletest.rs

```diff
@@ -274,13 +274,20 @@ fn main() -> Result<()> {
 fn run_dep_mode(target: String, mut args: impl Iterator<Item = OsString>) -> Result<()> {
     let path = args.next().expect("./miri run-dep must be followed by a file name");
     let mut config = test_config(&target, "", Mode::Yolo, /* with dependencies */ true);
-    config.program.args.remove(0); // remove the `--error-format=json` argument
-    config.program.args.push("--color".into());
-    config.program.args.push("always".into());
-    let mut cmd = ui_test::test_command(config, Path::new(&path))?;
-    // Separate the arguments to the `cargo miri` invocation from
-    // the arguments to the interpreted prog
-    cmd.arg("--");
+    config.program.args.clear(); // We want to give the user full control over flags
+    config.build_dependencies_and_link_them()?;
+
+    if let Ok(extra_flags) = env::var("MIRIFLAGS") {
+        for flag in extra_flags.split_whitespace() {
+            config.program.args.push(flag.into());
+        }
+    }
+    let mut cmd = config.program.build(&config.out_dir);
+
+    cmd.arg("--color").arg("always");
+    cmd.arg(path);
+
     cmd.args(args);
+    println!("Running {cmd:?}");
```

Comment:
I re-added this, as this is what `./miri run` also prints

Reply:
Well, for `./miri run` it doesn't contain 3 lines of linker junk though.^^

Reply:
I could print before adding that 😆

Reply:
`./miri run` prints only because that is forced behavior by cargo, I don't think we want to reproduce that here... maybe just add a `-q` in `./miri run`?

Reply:
Did that for both `run` and `run-dep`